src/vm/appdomain.cpp

```diff
@@ -12583,6 +12583,8 @@ AppDomain::RaiseAssemblyResolveEvent(
             if ( pSpec->IsIntrospectionOnly() 
 #ifdef FEATURE_FUSION
                     || pSpec->GetParentLoadContext() == LOADCTX_TYPE_UNKNOWN
+#elif FEATURE_CORECLR
+                    || pSpec->GetBindingContextFromParentAssembly(this) != GetTPABinderContext()
 #endif
```

Comment:
Per implementation in GetBindingContextFromParentAssembly, we would return TPABinderContext if the AssemblySpec does not contain ParentAssembly reference. Why does "Assembly.Load" return NULL? AssemblySpec for it should contain reference to the assembly triggering the load.